Makefile

```diff
@@ -97,10 +97,50 @@ build-release-tarballs: ## Create a series of `.tar.gz` files in the BIN_DIR dir
 	$(MAKE) build-x86_64-pc-windows-gnu
 	$(call tarball_release_binary,"x86_64-pc-windows-gnu","trin.exe","")
 
+##@ Tests
+
+EF_TESTS_TARGET = mainnet.tar.gz
+EF_TESTS_DIR = ./testing/ef-tests/mainnet
+LATEST_RELEASE_URL = https://api.github.com/repos/ethereum/consensus-spec-tests/releases/latest
+
+download_test_data:
+	@if [ -d $(EF_TESTS_DIR) ]; then \
+		echo "$(EF_TESTS_DIR) already downloaded. Skipping download."; \
+	else \
+		echo "Fetching the latest release URL for $(EF_TESTS_TARGET)..."; \
+		curl -s $(LATEST_RELEASE_URL) \
+		| grep "browser_download_url.*$(EF_TESTS_TARGET)" \
+		| cut -d : -f 2,3 \
+		| tr -d \" \
+		| wget -qi -; \
+		echo "$(EF_TESTS_TARGET) downloaded successfully."; \
+	fi
+
+extract_test_data: download_test_data
+	@if [ -d $(EF_TESTS_DIR) ]; then \
+		echo "$(EF_TESTS_DIR) already exists. Skipping extraction."; \
+	else \
+		echo "Extracting $(EF_TESTS_TARGET) into $(EF_TESTS_DIR)..."; \
+		mkdir -p $(EF_TESTS_DIR); \
+		tar -xzf $(EF_TESTS_TARGET) -C $(EF_TESTS_DIR); \
+		rm -f $(EF_TESTS_TARGET); \
+		echo "Extraction complete."; \
+	fi
+
+.PHONY: ef-tests
+ef-tests: extract_test_data ## Runs Ethereum Foundation tests.
+	@echo "Running tests..."
+	@cargo test -p ef-tests --features ef-tests
+	@echo "Tests complete."
+
+.PHONY: test
+test: ## Runs workspace tests.
```

Comment:
What do you think about having a make command like `test-workspace` (or `test-trin`?), where we will run `cargo test --workspace -- --nocapture` and then having `make test`  to run all tests (`ef-tests` + `test-workspace`)?

Reply:
I wouldn't want `make test` to run `ef-tests` as it downloads files

neither lighthouse or reth uses `make test` to run `ef-tests`

Lighthouse puts `ef-tests` behind `make test-full`, I will do that as then it is more intentional you want to run the expanded test suite, which would involve downloading files.

Both lighthouse and reth use `make test` for testing the workspace

Reply:
ok fair enough 👍 .